scripts/generate_book.py

```diff
@@ -233,10 +259,19 @@ def _copy_non_content_files():
         # Front-matter YAML
         yaml_fm = []
         yaml_fm += ['---']
-        yaml_fm += ['redirect_from:']
-        yaml_fm += ['  - "{}"'.format(_prepare_url(url_page).replace('_', '-').lower())]
-        if ix_file == 0:
-            yaml_fm += ['  - "/"']
+        # In case pre-existing links are sanitized
+        sanitized = url_page.lower().replace('_', '-')
+        if sanitized != url_page:
```

Comment:
ok one more annoying gotcha, here it would also fail if people used all lower-case names, but also used underscores. E.g.

if the page name is `my_page.html`

then the sanitized is `my-page.html`

then the `sanitized != url_page` is `True` and the case check will raise the error.

I think we could get around this w/ the following suggestion:

```suggestion
        if sanitized != url_page.replace('_', '-'):
```

Reply:
thanks for fixing these up @matthew-brett - sorry for the slow response but I was off at the neurips conference and just got back yesterday!

Reply:
I _think_ that's what the check is for on the next line - no?  I mean, in that case, `url_page.lower() != sanitized` and it won't raise the error.  Or am I not thinking straight?

Reply:
ohhh you're right. In my example above, the first check (`sanitized != url_page` would be True, but the second check `url_page.lower() == sanitized` would be False so the error wouldn't be triggered...in that case I think this LGTM! Merging away, thanks for grinding through this one w/ me :-)